common/scala/src/main/scala/whisk/core/entity/InvokerInstanceId.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.entity
+
+import spray.json.DefaultJsonProtocol
+import whisk.core.entity.ControllerInstanceId.LEGAL_CHARS
+import whisk.core.entity.ControllerInstanceId.MAX_NAME_LENGTH
+
+case class InvokerInstanceId(val instance: Int, name: Option[String] = None) {
+  def toInt: Int = instance
+}
+
+case class ControllerInstanceId(val asString: String) {
+  require(
+    asString.length <= MAX_NAME_LENGTH && asString.matches(LEGAL_CHARS),
+    "Controller instance id contains invalid characters")
```

Comment:
I still dislike the regex matching here, IIRC the ControllerInstanceId is passed via an ActivationMessage, so this is checked over and over again in serdes for no real value at all. On top of that, it's a concern of kafka, not of the instanceId per-se.